packages/realm/src/Dictionary.ts

```diff
@@ -57,6 +57,7 @@ const PROXY_HANDLER: ProxyHandler<Dictionary> = {
       internal.insertAny(prop, toBinding(value, undefined));
       return true;
     } else {
+      assert(typeof prop != "symbol", "Symbols cannot be used as keys of a dictionary");
```

Comment:
```suggestion
      assert(typeof prop !== "symbol", "Symbols cannot be used as keys of a dictionary");
```

Reply:
I'm a bit surprised the [`eqeqeq`](https://eslint.org/docs/latest/rules/eqeqeq) lint rule isn't enabled by the recommended settings.